C/tasks/north/sending_process/sending_process.cpp

```diff
@@ -0,0 +1,441 @@
+
+// Load data from storage
+static void loadDataThread(SendingProcess *loadData);
+// Send data from historian
+static void sendDataThread(SendingProcess *sendData);
+
+int main(int argc, char** argv)
+{
+	string pluginName = PLUGIN_NAME;
+
+	try
+	{
+		// Instantiate SendingProcess class
+		SendingProcess sendingProcess(argc, argv);
+
+		if (!sendingProcess.loadPlugin(pluginName))
+		{
+			Logger::getLogger()->fatal("Failed to load north plugin '%s'.", pluginName.c_str());
+			exit(2);
+		}
+
+		// Build JSON merged configuration (sendingProcess + pluginConfig
+		string mergedConfiguration("{ ");
+		// Get plugin default config
+		mergedConfiguration.append(sendingProcess.m_plugin->config());
+		mergedConfiguration += ", ";
+		mergedConfiguration.append(sendingProcessConfiguration);
+		mergedConfiguration += " }";
+
+		// Init plugin with merged configuration
+		sendingProcess.m_plugin->init(mergedConfiguration);
+
+                // Fetch last_object sent from foglamp.streams
+		if (!sendingProcess.getLastSentReadingId())
+		{
+			cerr << ">>>>>> Last object id for stream " << sendingProcess.getStreamId() << " NOT found, abort" << endl;
+                        exit(3);
+		}
+
+		// Launch the load thread
+		sendingProcess.m_thread_load = new thread(loadDataThread, &sendingProcess);
+		// Launch the send thread
+		sendingProcess.m_thread_send = new thread(sendDataThread, &sendingProcess);
+
+		// Check running time && handle signals
+		// Simulation with sleep
+		sleep(MAX_EXECUTION_TIME);
+
+		// End processing
+		sendingProcess.stop();
+	}
+	catch (const std::exception & e)
+	{
+		cerr << "Exception in " << argv[0] << " : " << e.what() << endl;
+
+		// Return failure for class instance
+		exit(1);
+	}
+
+	// Return success
+	exit(0);
+}
+
+/**
+ * Thread to load data from the storage layer.
+ *
+ * @param loadData    pointer to SendingProcess instance
+ */
+static void loadDataThread(SendingProcess *loadData)
+{
+        unsigned int    readIdx = 0;
+
+        while (loadData->isRunning())
+        {
+                if (readIdx >= DATA_BUFFER_ELMS)
+                {
+                        readIdx = 0;
+                }
+
+		/**
+		 * Check whether m_buffer[readIdx] is NULL or contains a ReadingSet
+		 *
+		 * Access is protected by a mutex.
+		 */
+                readMutex.lock();
+                ReadingSet *canLoad = loadData->m_buffer.at(readIdx);
+                readMutex.unlock();
+
+                if (canLoad)
+                {
+                        Logger::getLogger()->info("-- loadDataThread: " \
+                                                  "(stream id %d), readIdx %u, buffer is NOT empty, waiting ...",
+                                                  loadData->getStreamId(),
+                                                  readIdx);
+
+			// Load thread is put on hold
+			unique_lock<mutex> lock(waitMutex);
+			cond_var.wait(lock);
+
+			Logger::getLogger()->info("-- loadDataThread is now ready");
```

Comment:
Please think about less verbose logging. We should not flood the log with messages under normal circumstances.

Reply:
yes, just WIP

Reply:
Working on logging